lisa/runner.py

```diff
@@ -302,10 +306,11 @@ def _start_loop(self) -> None:
                             # current runner may not be done, but it doesn't
                             # have task temporarily. The root runner can start
                             # tasks from next runner.
+                            self._log.debug(
+                                f"No task available for runner : {runner.id}"
+                            )
                             break
-                        if not task_manager.has_idle_worker():
-                            has_idle_worker = False
-                            break
+                        has_idle_worker = task_manager.has_idle_worker()
```

Comment:
we can check the has_idle_worker every time, the variable may waste the idle workers.